crates/burn-core/src/nn/rnn/lstm.rs

```diff
@@ -353,4 +435,183 @@ mod tests {
         // Asserts that the gradients exist and are non-zero
         assert!(*some_gradient.any().into_data().value.first().unwrap());
     }
+
+    #[test]
+    fn test_bidirectional() {
+        TestBackend::seed(0);
+        let config = BiLstmConfig::new(2, 4, true);
+        let device = Default::default();
+        let mut lstm = config.init(&device);
+
+        fn create_gate_controller<const D1: usize, const D2: usize>(
+            input_weights: [[f32; D1]; D2],
+            input_biases: [f32; D1],
+            hidden_weights: [[f32; D1]; D1],
+            hidden_biases: [f32; D1],
+            device: &<TestBackend as Backend>::Device,
```

Comment:
Just a detail, but we can use `&burn_tensor::Device<TestBackend>` instead of this notation, just a bit prettier, especially when importing `burn_tensor::Device`.